enhancements/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,141 @@
+---
+title: openstack-baremetal-compute
+authors:
+  - "@pierreprinetti"
+reviewers:
+  - "@openshift/team-openstack-cloud-provider"
+approvers:
+  - TBD
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: provisional
+
+---
+
+# Openstack Baremetal Compute
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenStack's [Nova][openstack-nova] service can be configured to provision
+baremetal machines as OpenShift Compute Nodes.
+
+This enhancement will allow [Ironic][openstack-ironic]-provisioned machines to
+be used as OpenShift Compute Nodes, when they are made available as Nova
+flavors.
+
+## Motivation
+
+In heavy-loaded clusters, or when the workload is better handled with specially
+purposed hardware, the functional requirement of controlling where the work is
+computed might outweight the added value of virtualisation.
+
+In OpenStack clusters where Nova transparently provisions baremetal machines,
+this enhancement will ensure that OpenShift installations can use them as
+Compute Nodes.
+
+### Goals
+
+* Enable running OpenShift on a mix of virtual and baremetal machines.
+* Baremetal machines can be added and removed from a running OpenShift cluster.
+
+### Non-Goals
+
+An OpenStack setting where baremetal machines can't be attached to a virtual
+subnet is not supported in this enhancement.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+* The reference OpenStack cluster transparently provisions baremetal machines
+* Provisioned baremetal machines can be attached to a Neutron virtual subnet
+
+### Probable issues
+
+* The Ignition boot system consisting on a second payload downloaded at runtime
+  requires the baremetal machine to have access to Swift early in the boot
+  process.
```

Comment:
Could you elaborate? How is this different from the VM case? So long as networking works as expected, this should work the same no matter if it's VM or BM.

What am I missing?

Reply:
You are not missing anything, it's just like VMs. I just wanted to outline issues that we might have to face in the implementation. Do you think it's misplaced here?

Reply:
I'll just remove this point, as it is not matching with the rest of the doc (which only touches Compute nodes as BM).

In case we go with the "general case" (i.e. "mixed installs" as opposed to "baremetal compute-nodes"), we will have to put this back